test/rules/callable-types/test.ts.lint

```diff
@@ -23,6 +23,11 @@ type T = {
     ~~~~~~~~~ [type % ('() => void')]
 };
 
+type T = T2 & {
+    (): void;
```

Comment:
what happens when there are multiple fields in this type literal? seems like it would remove all the semicolons, which could break syntax again